declarative/codegen/src/main/java/io/helidon/declarative/codegen/validation/ValidatedTypeGenerator.java

```diff
@@ -0,0 +1,480 @@
+        List<Property> recordComponents = new ArrayList<>();
+
+        type.elementInfo()
+                .stream()
+                .filter(it -> it.kind() == ElementKind.RECORD_COMPONENT)
+                .filter(it -> needsWork(constraintAnnotations, it))
+                .forEach(element -> {
+                    String propertyName = element.elementName();
+                    Property property = new Property(propertyName, "check" + capitalize(propertyName), element.typeName());
+                    recordComponents.add(property);
+
+                    checkMethod.addContent("validation__res = validation__res.merge(")
+                            .addContent(property.checkMethodName())
+                            .addContent("(validation__ctx, validation__instance.")
+                            .addContent(propertyName)
+                            .addContentLine("()));");
+
+                    addCheckMethod(generatedType,
+                                   classModel,
+                                   constraintAnnotations,
+                                   fieldHandler,
+                                   "RECORD_COMPONENT",
+                                   element,
+                                   propertyName);
+                });
+
+        checkMethod.addContentLine();
+
+        addCheckWithPropertyNameMethods(classModel, type.typeName(), recordComponents);
+    }
+
+    private void processValidatedClass(TypeName generatedType,
+                                       Collection<TypeName> constraintAnnotations,
+                                       TypeInfo type,
+                                       ClassModel.Builder classModel,
+                                       FieldHandler fieldHandler,
+                                       Method.Builder checkMethod) {
+
+        List<Property> properties = new ArrayList<>();
+
+        // non-private non-static methods that match getter pattern (we only add those annotated with a constraint or Valid)
+        type.elementInfo()
+                .stream()
+                .filter(ElementInfoPredicates::isMethod)
+                .filter(Predicate.not(ElementInfoPredicates::isPrivate))
+                .filter(Predicate.not(ElementInfoPredicates::isStatic))
+                .filter(ElementInfoPredicates::hasNoArgs)
+                .filter(Predicate.not(ElementInfoPredicates::isVoid))
+                .filter(it -> needsWork(constraintAnnotations, it))
+                .forEach(element -> {
+                    String propertyName = element.elementName();
+                    if (propertyName.startsWith("get") && propertyName.length() > 3) {
+                        propertyName = propertyName.substring(3);
+                        if (propertyName.length() == 1) {
+                            propertyName = propertyName.toLowerCase(Locale.ROOT);
+                        } else if (!Character.isUpperCase(propertyName.charAt(1))) {
```

Comment:
I think this is a mistake. You wanted the character at the index 0, right?

Reply:
This is not redundant, nor is it a mistake - this is for cases such as `getADSL` - we want to keep all capital...